src/Tizen.NUI.WindowSystem/src/public/InputGesture.cs

```diff
@@ -0,0 +1,557 @@
+        public void ReleasePalmCoverData(IntPtr data)
+        {
+            if (data == IntPtr.Zero)
+            {
+                throw new ArgumentException("palmCoverData is not valid.");
+            }
+            Interop.InputGesture.ErrorCode res = Interop.InputGesture.PalmCoverFree(_handler, data);
+            ErrorCodeThrow(res);
+            Log.Debug(LogTag, "ReleasePalmCoverData");
+        }
+
+        /// <summary>
+        /// Grabs a global gesture
+        /// </summary>
+        /// <param name="data">gesture data to grab</param>
+        /// <exception cref="ArgumentException">Thrown when failed of invalid argument.</exception>
+        public void GrabGesture(IntPtr data)
+        {
+            if (data == IntPtr.Zero)
+            {
+                throw new ArgumentException("gesture data is not valid.");
+            }
+            Interop.InputGesture.ErrorCode res = Interop.InputGesture.GestureGrab(_handler, data);
+            ErrorCodeThrow(res);
+            Log.Debug(LogTag, "GrabGesture");
+        }
+
+        /// <summary>
+        /// Ungrabs a global gesture.
+        /// </summary>
+        /// <param name="data">gesture data to ungrab</param>
+        /// <exception cref="ArgumentException">Thrown when failed of invalid argument.</exception>
+        public void UngrabGesture(IntPtr data)
+        {
+            if (data == IntPtr.Zero)
+            {
+                throw new ArgumentException("gesture data is not valid.");
+            }
+            Interop.InputGesture.ErrorCode res = Interop.InputGesture.GestureUngrab(_handler, data);
+            ErrorCodeThrow(res);
+            Log.Debug(LogTag, "UngrabGesture");
+        }
+
+        /// <summary>
+        /// Emits the event when the edge swipe event comes
+        /// </summary>
+        /// <exception cref="ArgumentException">Thrown when failed of invalid argument.</exception>
+        public event EventHandler<EdgeSwipeEventArgs> EdgeSwipeEventHandler
+        {
+            add
+            {
+                if (_edgeSwipeEventHandler == null)
+                {
+                    _edgeSwipeDelegate = (IntPtr userData, int mode,  int fingers, int sx, int sy, int edge) =>
+                    {
+                        EdgeSwipeEventArgs args = new EdgeSwipeEventArgs(mode, fingers, sx, sy, edge);
+                        Log.Debug(LogTag, "EdgeSwipe Event received. mode: " + mode + ", fingers: " + fingers);
+                        _edgeSwipeEventHandler?.Invoke(null, args);
+                    };
+                    Interop.InputGesture.ErrorCode res = Interop.InputGesture.SetEdgeSwipeCb(_handler, _edgeSwipeDelegate, IntPtr.Zero);
+                    ErrorCodeThrow(res);
+                }
+
+                _edgeSwipeEventHandler += value;
+            }
+            remove
+            {
+                _edgeSwipeEventHandler -= value;
```

Comment:
It seems that there's only code for setting the callback, but no code for unsetting it. In case where the C# object gets disposed while the callback delegate has been set, an error might occur when the EFL callback is invoked without the corresponding callback function. Therefore, it appears necessary to include code for removing the EFL callback during event removal, as well as code to unset the callback when the InputGesture object is being disposed.

Reply:
Thank you for your comment. I added the code snippet unsetting callback function.